File: Java/opentraceability/src/main/java/opentraceability/models/common/StandardBusinessDocumentHeader.java
package opentraceability.models.common;

import opentraceability.Constants;
import opentraceability.utility.attributes.OpenTraceabilityAttribute;
import opentraceability.utility.attributes.OpenTraceabilityObjectAttribute;
import java.time.OffsetDateTime;

public class StandardBusinessDocumentHeader {

    @OpenTraceabilityAttribute(ns = Constants.SBDH_NAMESPACE, name = "HeaderVersion")
    public String HeaderVersion = "";

    @OpenTraceabilityObjectAttribute
    @OpenTraceabilityAttribute(ns = Constants.SBDH_NAMESPACE, name = "Sender")
    public SBDHOrganization Sender;

    @OpenTraceabilityObjectAttribute
    @OpenTraceabilityAttribute(ns = Constants.SBDH_NAMESPACE, name = "Receiver")
    public SBDHOrganization Receiver;

    @OpenTraceabilityObjectAttribute
    @OpenTraceabilityAttribute(ns = Constants.SBDH_NAMESPACE, name = "DocumentIdentification")
    public SBDHDocumentIdentification DocumentIdentification;

    public static StandardBusinessDocumentHeader DummyHeader = new StandardBusinessDocumentHeader();
    static {
        DummyHeader.HeaderVersion = "x";

        DummyHeader.Sender = new SBDHOrganization();
        DummyHeader.Sender.Identifier = "x";
        DummyHeader.Sender.ContactInformation = new SBDHContact();
        DummyHeader.Sender.ContactInformation.ContactName = "x";
        DummyHeader.Sender.ContactInformation.EmailAddress = "x";

        DummyHeader.Receiver = new SBDHOrganization();
        DummyHeader.Receiver.Identifier = "x";
        DummyHeader.Receiver.ContactInformation = new SBDHContact();
        DummyHeader.Receiver.ContactInformation.ContactName = "x";
        DummyHeader.Receiver.ContactInformation.EmailAddress = "x";

        DummyHeader.DocumentIdentification = new SBDHDocumentIdentification();
        DummyHeader.DocumentIdentification.CreationDateAndTime = OffsetDateTime.now();
        DummyHeader.DocumentIdentification.InstanceIdentifier = "x";
        DummyHeader.DocumentIdentification.MultipleType = "false";
        DummyHeader.DocumentIdentification.Standard = "x";
        DummyHeader.DocumentIdentification.Type = "x";
        DummyHeader.DocumentIdentification.TypeVersion = "3.0";
    }
}

File: Java/OpenTraceability/src/main/kotlin/opentraceability/models/common/standardbusinessdocumentheader.kt
package models.common

import Constants
import utility.attributes.*
import java.time.OffsetDateTime

class StandardBusinessDocumentHeader {

    @OpenTraceabilityAttribute(Constants.SBDH_NAMESPACE, "HeaderVersion")
    var HeaderVersion: String = ""

    @OpenTraceabilityObjectAttribute
    @OpenTraceabilityAttribute(Constants.SBDH_NAMESPACE, "Sender")
    var Sender: SBDHOrganization? = null

    @OpenTraceabilityObjectAttribute
    @OpenTraceabilityAttribute(Constants.SBDH_NAMESPACE, "Receiver")
    var Receiver: SBDHOrganization? = null

    @OpenTraceabilityObjectAttribute
    @OpenTraceabilityAttribute(Constants.SBDH_NAMESPACE, "DocumentIdentification")
    var DocumentIdentification: SBDHDocumentIdentification? = null

    companion object {
        var DummyHeader: StandardBusinessDocumentHeader = StandardBusinessDocumentHeader()
            get() {
                var dh: StandardBusinessDocumentHeader = StandardBusinessDocumentHeader()
                dh.HeaderVersion = "x";

                dh.Sender = SBDHOrganization()
                dh.Sender!!.Identifier = "x"
                dh.Sender!!.ContactInformation = SBDHContact()
                dh.Sender!!.ContactInformation.ContactName = "x"
                dh.Sender!!.ContactInformation.EmailAddress = "x"

                dh.Receiver = SBDHOrganization()
                dh.Receiver!!.Identifier = "x"
                dh.Receiver!!.ContactInformation = SBDHContact()
                dh.Receiver!!.ContactInformation.ContactName = "x"
                dh.Receiver!!.ContactInformation.EmailAddress = "x"

                dh.DocumentIdentification = SBDHDocumentIdentification()
                dh.DocumentIdentification!!.CreationDateAndTime = OffsetDateTime.now()
                dh.DocumentIdentification!!.InstanceIdentifier = "x"
                dh.DocumentIdentification!!.MultipleType = "false"
                dh.DocumentIdentification!!.Standard = "x"
                dh.DocumentIdentification!!.Type = "x"
                dh.DocumentIdentification!!.TypeVersion = "3.0"

                return dh
            }
    }
}
